p11-kit/test-proxy.c

```diff
@@ -192,30 +192,25 @@ test_initialize_child (void)
 
 struct {
 	char *directory;
-	const char *system_file;
-	const char *system_modules;
 } test;
 
-extern const char *p11_config_system_file;
-extern const char *p11_config_system_modules;
-
 static void
 setup (void *unused)
 {
 	test.directory = p11_test_directory ("test-proxy");
-	test.system_file = p11_config_system_file;
-	p11_config_system_file = SRCDIR "/p11-kit/fixtures/test-system-none.conf";
-	test.system_modules = p11_config_system_modules;
-	p11_config_system_modules = test.directory;
+	p11_kit_override_system_files (SRCDIR "/p11-kit/fixtures/test-system-none.conf",
+				       NULL,
+				       SRCDIR "/p11-kit/fixtures/nonexistent",
+				       test.directory,
+				       SRCDIR "/p11-kit/fixtures/nonexistent");
 }
 
 static void
 teardown (void *unused)
 {
 	p11_test_directory_delete (test.directory);
 	free (test.directory);
-	p11_config_system_file = test.system_file;
-	p11_config_system_modules = test.system_modules;
+	p11_kit_override_system_files (NULL, NULL, NULL, NULL, NULL);
```

Comment:
Yes I agree this is confusing, though I am not sure if it's worth adding a new function as `p11_kit_override_system_files` was introduced for testing purposes. I guess maybe it would be safer to get back to the old way (i.e., setting `p11_config_` variables manuall) for this PR; I will update it soon.